engine/src/test/java/io/zeebe/engine/processor/workflow/CatchEventBehaviorCorrelationKeyExtractionResultHandlerTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.zeebe.el.EvaluationResult;
+import io.zeebe.el.ResultType;
+import io.zeebe.engine.processor.workflow.CatchEventBehavior.CorrelationKeyExtractionResultHandler;
+import io.zeebe.engine.processor.workflow.message.MessageCorrelationKeyContext;
+import io.zeebe.engine.processor.workflow.message.MessageCorrelationKeyException;
+import java.util.Locale;
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CatchEventBehaviorCorrelationKeyExtractionResultHandlerTest {
+
+  public static final String TEST_EXPRESSION = "expression";
+  @Mock MessageCorrelationKeyContext mockMessageCorrelationKeyContext;
+
+  CorrelationKeyExtractionResultHandler sutResultHandler;
+
+  Locale defaultLocaleToRestore;
+
+  @Before
+  public void setUp() {
+    sutResultHandler = new CorrelationKeyExtractionResultHandler(mockMessageCorrelationKeyContext);
+    defaultLocaleToRestore = Locale.getDefault();
+    Locale.setDefault(Locale.US);
```

Comment:
It was needed because one of the error messages formats the value that was the result of the evaluation expression. This value is formatted locale-dependent.

I now removed this from the test and instead set the locale where the message is generated:
`String.format(Locale.US, "Failed to extra...`